app.go

```diff
@@ -918,28 +918,41 @@ func withTimeout(ctx context.Context, param *withTimeoutParams) error {
 		if err != context.DeadlineExceeded {
 			return err
 		}
-		// On timeout, report running hook's caller and recorded
-		// runtimes of hooks successfully run till end.
-		r := param.lifecycle.hookRecords()
-		sort.Sort(r)
-		caller := param.lifecycle.runningHookCaller()
-		// TODO: Once this is integrated into fxevent, we can
-		// leave error unchanged and send this to fxevent.Logger, whose
-		// implementation can then determine how the error is presented.
-		if len(r) > 0 {
-			return fmt.Errorf("%v hook added by %v failed: %w\n%+v",
-				param.hook,
-				caller,
-				err,
-				r)
+		return formatTimeoutError(err, param)
```

Comment:
We can dedupe this a bit:

```
var err error
select {
case <-ctx.Done():
  err = ctx.Err()
case err = <-c:
}

if err != context.DeadlineExceeded {
  return err
}

return formatTimeoutError(err, param)
```

(at which point this can go back to inlining formatTimeoutError into the withTimeout body.)